gapic-generator-java/src/main/java/com/google/api/generator/gapic/protoparser/Parser.java

```diff
@@ -1000,6 +1034,21 @@ private static Field parseField(
       }
     }
 
+    // If field is annotated with fieldInfo.format = UUID4 and fieldBehavior != REQUIRED, then
+    // autopopulate the field
+    // TODO: Autopopulation requires fieldInfo.formatValue = UUID4; when more types are supported,
+    // this logic will need to be updated
+    if (fieldOptions.hasExtension(FieldInfoProto.fieldInfo)) {
+      fieldInfoFormat = fieldOptions.getExtension(FieldInfoProto.fieldInfo).getFormat();
+      if (fieldOptions.getExtension(FieldInfoProto.fieldInfo).getFormatValue() == 1) {
```

Comment:
have refactored it slightly such that now it just sets `Field.isRequired` and `Field.fieldInfoFormat`. I think the logic of auto-population should exist on the `AbstractTransportServiceStubClassComposer.java` (e.g. it checks that the field name is listed in `Method.autoPopulatedFields()`, thefield is not required, and the field format is UUID4 before auto-generating it). 

`FieldTest.java` doesn't currently exist - I'm not sure if it makes sense to add given that the unit tests for `Field` would be relatively trivial. But thought I'd ask given our earlier convo on tests :)

Reply:
> FieldTest.java doesn't currently exist - I'm not sure if it makes sense to add given that the unit tests for Field would be relatively trivial

I agree, if it's just getters/setters, we don't have to create unit tests for it. There are different opinions around it, like this [discussion](https://stackoverflow.com/questions/6197370/should-unit-tests-be-written-for-getter-and-setters), but I don't think we should do it. 

However, I see that there are changes in hashcode/equals as well, which usually need to be tested if they are handwritten. And I'm not sure why we [added them manually](https://github.com/googleapis/sdk-platform-java/pull/622) in the first place instead of using the hashcode/equals provided by [Autovalue](https://yaqs.corp.google.com/eng/q/4616341951873024). Let's discuss it offline tomorrow.

Reply:
added them to hashcode() per our offline discussion yesterday and added additional unit tests to ParserTest